mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -634,12 +634,13 @@ struct S3Message {
 }
 
 impl S3Message {
-    /// Add a header to this message.
-    fn add_header(
+    /// Add a header to this message. The header is added if necessary and any existing values for
+    /// this header are removed.
+    fn set_header(
```

Comment:
Was there any use of adding a new header with different value rather than overwriting it, earlier?

Reply:
Nope, that's why I changed it. Multiple headers with the same name is a pretty niche HTTP thing that we don't need.